File: tests/cases/check/frozen-rules/csharp/inputs/src/Orders/Billing.cs
using CodeConstraints.Rules;

namespace Orders;

// [Sealed] deleted. [Immutable] survives untouched.
[Immutable]
public class Receipt
{
    public decimal Total { get; }

    public Receipt(decimal total) { Total = decimal.Round(total, 2); }
}

// The allow-list grew: same tag, different params - `weakened`.
[NoInstantiation(Allow = new[] { "AuditEntry", "Receipt" })]
public class CheckoutService
{
    // [NoSideEffects] deleted from an *operation*.
    public decimal Subtotal(int n) { return n; }
}

File: tests/cases/check/frozen-rules/python/inputs/src/orders/billing.py
from cdec_rules import immutable, no_instantiation, sealed


@immutable
class Receipt:
    """`@sealed` deleted. `@immutable` survives untouched, so only one tag
    fires here."""

    def __init__(self, total: float) -> None:
        self.total: float = round(total, 2)


@no_instantiation(allow=["AuditEntry", "Receipt"])
class CheckoutService:
    """The allow-list grew: same tag name, different params - `weakened`."""

    def subtotal(self, n: int) -> float:
        """`@no_side_effects` deleted from an *operation*."""
        return float(n)
